### PR TITLE
13425 team picker bug

### LINKED ISSUE #13425: TeamPicker cannot select team if not included in initial search
* Lower the search query limit, https://github.com/grafana/grafana/blob/master/public/app/core/components/Picker/TeamPicker.tsx#L42 
* Search for team, try to select it
* Team is not selected (unless part of the initial list)

### PR DESCRIPTION
**Pull request by jschill**
There's a problem with the build so this is not to be merged, but feel free to start reviewing the code

Verify:
* UserPicker
* TeamPicker
* DescriptionPicker
* "DatasourcePicker" (in Explore)
* Light/dark mode
* Need input re css class namings
* Tag filtering in "Search"

fixes #13425 

Note that this PR will update to use the new build container (with the bumped node-version).

### CODE REVIEW

File: public/app/core/components/PermissionList/AddPermission.tsx [hunk 1]
```diff
@@ -50,11 +50,11 @@ class AddPermissions extends Component<Props, NewDashboardAclItem> {
   };
 
   onUserSelected = (user: User) => {
-    this.setState({ userId: user ? user.id : 0 });
+    this.setState({ userId: user && !Array.isArray(user) ? user.id : 0 });
```
Comment (jschill) [inline public/app/core/components/PermissionList/AddPermission.tsx]: With new react-select, user can come in as an empty array.

File: public/app/core/components/PermissionList/AddPermission.tsx [hunk 2]
```diff
@@ -107,31 +106,22 @@ class AddPermissions extends Component<Props, NewDashboardAclItem> {
 
             {newItem.type === AclTarget.User ? (
               <div className="gf-form">
-                <UserPicker
-                  onSelected={this.onUserSelected}
-                  value={newItem.userId.toString()}
-                  className={pickerClassName}
-                />
+                <UserPicker onSelected={this.onUserSelected} className={pickerClassName} />
```
Comment (jschill) [inline public/app/core/components/PermissionList/AddPermission.tsx]: New react-select is stateful, so we don't have to manage the value-state ourselves anymore (unless we really need to do that)

File: public/app/core/components/Picker/ResetStyles.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  menuList: () => ({}),
+  multiValue: () => ({}),
+  multiValueLabel: () => ({}),
+  multiValueRemove: () => ({}),
+  noOptionsMessage: () => ({}),
+  option: () => ({}),
+  placeholder: () => ({}),
+  singleValue: () => ({}),
+  valueContainer: () => ({}),
+};
```
Comment (jschill) [inline public/app/core/components/Picker/ResetStyles.tsx]: Can we do this smarter? Any ideas?

File: public/sass/components/_form_select_box.scss [hunk 4]
```diff
@@ -29,73 +15,110 @@ $select-option-selected-bg: $dropdownLinkBackgroundActive;
...
+  display: none;
+}
 
-  .Select-menu-outer {
-    border: 0;
-    width: auto;
+.gf-form-select-box__value-container {
+  display: table-cell;
+  padding: 8px 10px;
+  > div {
```
Comment (jschill) [inline public/sass/components/_form_select_box.scss]: Can we do this better?

File: public/app/core/components/Picker/ResetStyles.tsx (hunk 3 above)
Comment (peterholmberg) [reply public/app/core/components/Picker/ResetStyles.tsx]: Such a weird way to do this from react-select point of view. Why couldn't it just be a bool?

Comment (jschill) [reply public/app/core/components/Picker/ResetStyles.tsx]: Good point. I will send them a PR.

**Pull request merged**